### PR TITLE
Feat:  Explore screen

### PR DESCRIPTION
**Pull request by lukewalczak**
- [x] - `BookingOneWay`
- [x] - `BookingMulticity`
- [x] - `BookingReturn`
- [x] - add missing graphql types

### CODE REVIEW

File: app/MMB/src/scenes/explore/Explore.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    switch (data.__typename) {
+      case 'BookingOneWay':
+        return <ExploreBookingOneWay data={data} />;
+      case 'BookingMulticity':
+        return <ExploreBookingMulticity data={data} />;
+      default:
+        return null;
+    }
+  }
+  return null;
```
Comment (tbergq) [inline app/MMB/src/scenes/explore/Explore.js]: This looks like a component. Please move it a separate file and have it fetch __typename in a fragment.

File: app/MMB/src/scenes/explore/ExploreBookingMulticity.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+const getDepartureAirportCity = leg =>
+  idx(leg, _ => _.departure.airport.city.name) || '';
+const getDepartureAirportCode = leg =>
+  idx(leg, _ => _.departure.airport.code) || '';
+const getDepartureAirportCountry = leg =>
+  idx(leg, _ => _.departure.airport.country.name);
+
+const ExploreBookingMulticity = (props: Props) => {
+  const trips = idx(props, _ => _.data.trips);
```
Comment (tbergq) [inline app/MMB/src/scenes/explore/ExploreBookingMulticity.js]: if you do -> `const trips = idx(props, _ => _.data.trips) || [];`
then you can skip this check `if (trips && trips.length > 0)` and just do `trips.map`

File: app/MMB/src/scenes/explore/ExploreBookingMulticity.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        );
+      });
+    }
+    return null;
+  };
+
+  return (
+    <ScrollView>
+      <TitledMenuGroup title={<Translation id="mmb.explore.airports" />}>
+        {renderAirportGroup()}
```
Comment (tbergq) [inline app/MMB/src/scenes/explore/ExploreBookingMulticity.js]: Also I would rather see that we use components. This you just uppercase this, it is a component.

File: app/MMB/src/scenes/explore/ExploreBookingMulticity.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      </TitledMenuGroup>
+      <TitledMenuGroup title={<Translation id="mmb.explore.destinations" />}>
+        {renderDestinationsGroup()}
```
Comment (tbergq) [inline app/MMB/src/scenes/explore/ExploreBookingMulticity.js]: same here

File: app/MMB/src/scenes/explore/ExploreBookingOneWay.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+const renderItem = (isAirport: boolean, title: string, description: string) => (
+  <MenuItem
+    title={<Translation passThrough={title} />}
+    description={<Translation passThrough={description} />}
+    icon={isAirport ? <TextIcon code="a" /> : <TextIcon code="c" />}
+    isActive={false}
+    onPress={warnTodo}
+  />
+);
```
Comment (tbergq) [inline app/MMB/src/scenes/explore/ExploreBookingOneWay.js]: This looks like the exact same renderItem as above, extract it to it's own file, and reuse it from both components

**Comment by ferrannp** [comment]
You have a lot of if-else and conditions there. I think we would really need unit tests for `ExploreAirportGroup.js` & `ExploreDestinationsGroup.js`.

**Comment by lukewalczak** [comment]
Totally agree. I will open an issue for that.

**Comment by tbergq** [comment]
skjermbilde 2018-08-07 kl 15 04 39

Also separators seems to be missing

File: schema.graphql [hunk 6]
```diff
@@ -2241,4 +2241,4 @@ type WhitelabeledServices {
   lounge: LoungeService
   parking: ParkingService
   carRental: CarRentalService
-}
+}
```
[the remark this answers is not in the document]
Comment (lukewalczak) [reply schema.graphql]: Weird...

File: app/MMB/src/scenes/explore/ExploreAirportGroup.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  const oneWayTrips = idx(data, _ => _.trip.legs) || [];
+
+  const inboundTrips = idx(data, _ => _.inbound.legs) || [];
+
+  const outboundTrips = idx(data, _ => _.outbound.legs) || [];
+
+  const airports = [];
+
+  switch (type) {
+    case 'BookingOneWay':
```
[the remark this answers is not in the document]
Comment (lukewalczak) [reply app/MMB/src/scenes/explore/ExploreAirportGroup.js]: Ok, will refactor.

**Pull request merged**